astroid/brain/brain_dataclasses.py

```diff
@@ -231,9 +232,11 @@ def infer_dataclass_attribute(
 
 
 def infer_dataclass_field_call(
-    node: AssignName, ctx: context.InferenceContext = None
-) -> Generator:
+    node: Call, ctx: Optional[context.InferenceContext] = None
+) -> Optional[Generator]:
     """Inference tip for dataclass field calls."""
+    if not isinstance(node.parent, (AnnAssign, Assign)):
+        return
```

Comment:
I think this should be `raise UseInferenceDefault` instead. Please test it before changing it as I'm not 100% certain.